client/components/board/boardFactory.js

```diff
@@ -1,22 +1,22 @@
 angular.module('OpenEXP')
     .factory('boardFactory', () => {
+
+        // requirements
         const OpenBCIBoard = require('openbci-sdk');
         const ourBoard = new OpenBCIBoard.OpenBCIBoard();
 
-        // wrote a method in the openbci-sdk to return an array of ports and
-        // wrapping it in a promise here
+        // return a promise to listen devices
         var listDevices = () => {
             return new Promise((res,rej) => {
               ourBoard.listPorts().then((value) => {
                 res(value)
-                console.log(value)
+              }).catch((err) => {
+                console.log("error!")
               })
```

Comment:
If it returns a Promise then we don't need to wrap it in a promise. Pretty sure I wrote it that way initially because it wasn't at the time, but that may have changed.